lib/Elastica/Response.php

```diff
@@ -183,7 +190,8 @@ public function getData()
                 $this->_error = true;
             } else {
                 try {
-                    $response = JSON::parse($response);
+                    $flag = $this->getJsonBigintConversion() ? JSON_BIGINT_AS_STRING : 0;
+                    $response = JSON::parse($response, false, 512, $flag);
```

Comment:
I would suggest the following change here:
```
if (bigint) {
   $response = JSON::parse($response, false, 512, JSON_...)
} else {
   $response = JSON::parse($response);
}
```
The main reason for this are the defaults. They are correct at the moment, but in case a PHP version changes these defaults, we would stay with the old ones. I think if we apply special flags, this is ok. But for the default we should hardcoded fewer flags.